protocols/core.js

```diff
@@ -84,7 +84,7 @@ export default class Core extends EventEmitter {
     state.queryPort = options.port
     // because lots of servers prefix with spaces to try to appear first
     state.name = (state.name || '').trim()
-    state.connect = `${state.gameHost || options.host || options.address}:${state.gamePort || options.port}`
+    state.connect = state.connect || `${state.gameHost || options.host || options.address}:${state.gamePort || options.port}`
```

Comment:
This is technically a breaking change and a bug, currently `discord`, `eldewrito` and `buildandshoot` do (or may) set this field manually and although the state should use that one, it does not and manually sets one (via the fields before editing).

It is a breaking change as users of these protocols could get different data after updating to a newer version.
And it is a bug as the field should respect the protocol's selection and not overwrite this.

Reply:
Talked over on discord and this should be fine.